portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -41,9 +44,13 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String name = getParameter(request, "name", "");
     String email = getParameter(request, "email", "");
     String comment = getParameter(request, "comment", "");
-    entries.add(name);
-    entries.add(email);
-    entries.add(comment);
+    Entity entryEntity = new Entity("Entry");
+    entryEntity.setProperty("name", name);
+    entryEntity.setProperty("email", email);
+    entryEntity.setProperty("comment", comment);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(entryEntity);
     response.sendRedirect("/index.html");
```

Comment:
It would be good to move the write to a separate method. so,

doPost() {
  writeToDatastore(getParameter("name", name), ...)
  response.sendRequest(...)
}

private void WriteToDatastore(String name, String email, String comment) {
  Entity entryEntity = ..
  ...
  datastore.put(entryEntity);
}

Reply:
Done